README.md

```diff
@@ -1,33 +1,57 @@
 
 *   [Build with Linux](#linux-instructions)
 *   [Build using OS X](#using-os-x)
-*   Using the stable [upstreamed](#using-the-stable-upstreamed-version) version
 *   [Debugging](#debugging)
 *   [Support](#support)
 *   [How to Contribute](#how-to-contribute)
 
 
-
 ## Linux instructions
+There are 3 ways to install nGraph-tf.
 
+| Option          | Build TF from Source   | Build nGraph from Source | Notes |
+|:---:|:---:|:---:|:---: |
+| [1](#option-1-build-ngraph-using-an-existing-tensorflow-installation) |   No        |  Yes         | |
+| [2](#option-2-build-ngraph-using-tensorflow-source) |  Yes        |  Yes         | Builds unit tests. Recommended for contributing to nGraph-Tensorflow |
+| [3](#option-3-build-tensorflow-source-with-ngraph) |  Yes        |  No          | nGraph enabled by default |
 
-### Option 1: Use an existing TensorFlow v1.11.0 (or greater) installation
+#### Create a python virtual environment
 
-1. You need to instantiate a specific kind of `virtualenv`  to 
-   be able to proceed with the `ngraph-tf` bridge installation. For 
-   systems with Python 3.n or Python 2.7, these commands are
+You need to instantiate a specific kind of `virtualenv`  to be able to proceed with the `ngraph-tf` installation. For Python 3.n or Python 2.7, do
 
         virtualenv --system-site-packages -p python3 your_virtualenv 
         virtualenv --system-site-packages -p /usr/bin/python2 your_virtualenv  
         source your_virtualenv/bin/activate # bash, sh, ksh, or zsh
+
+Typically the following python packages are also needed `numpy mock keras keras_applications`.
+
+        pip install -U numpy mock keras keras_applications
+
+Note: Depending on the version of Python and the packages already installed on your system,the above list may vary.
+
+#### Install bazel for building TensorFlow Source
+
+The installation prerequisites are the same as described in the TensorFlow [prepare environment] for linux.
+
+1. We use the standard build process called "bazel". The instructions were tested with [bazel version 0.16.0]. 
+
+        wget https://github.com/bazelbuild/bazel/releases/download/0.16.0/bazel-0.16.0-installer-linux-x86_64.sh      
+        chmod +x bazel-0.16.0-installer-linux-x86_64.sh
+        ./bazel-0.16.0-installer-linux-x86_64.sh --user
+
+2. Add and source the ``bin`` path to your ``~/.bashrc`` file in order to call bazel from the installation we set up:
+
+        export PATH=$PATH:~/bin
+        source ~/.bashrc 
+
+### Option 1: Build nGraph using an existing TensorFlow installation
+
+1. Create and activate a [python virtual environment](#create-a-python-virtual-environment)
     
 2. Install TensorFlow v1.11.0. Note that this is a pre-release so you need 
    to use the following steps to install this:
```

Comment:
As v1.11.0 is official release. We may not need this line "Note that this is a pre-release so you need 
   to use the following steps to install this:"

Reply:
Agree.